docs/source/tutorials/chat.rst

```diff
@@ -0,0 +1,164 @@
+=================================
+Fine-tuning LLaMA3 with chat data
+=================================
+
+In this tutorial, we'll demystify what prompt templates are and when you'll need them
+and talk about the differences between prompt templating for LLaMA2 and LLaMA3. Then,
+we'll wrap up with a LLaMA3 finetuning example on a custom chat dataset.
+
+.. grid:: 2
+
+    .. grid-item-card:: :octicon:`mortar-board;1em;` You will learn:
+
+      * What is a prompt template and how is it different from special tokens
+      * When to use and not to use a prompt template
+      * How to use your own chat dataset to fine-tune LLaMA3
+
+    .. grid-item-card:: :octicon:`list-unordered;1em;` Prerequisites
+
+      * Be familiar with :ref:`configuring datasets<dataset_tutorial_label>`
+      * Know how to :ref:`download LLaMA3 weights <llama3_label>`
+
+
+Prompt templates & special tokens
+---------------------------------
+
+Prompt templates and tokenization schemes are often conflated, and it's sometimes
+unclear what you need to do to format your data to optimize the training performance
+of your model. Let's walk through the LLaMA2/LLaMA3 templates to better understand
+the distinction.
+
+In torchtune, a prompt template is simply a structured way to format your prompt
+into the model. The prompt template adds flavor text to prime the model to expect
+a certain structure when it is used for inference. From the `official LLaMA2 prompt
+template guide <https://llama.meta.com/docs/model-cards-and-prompt-formats/meta-llama-2>`_
+for the LLaMA2 chat model, we can see that special tags are added:
+
+.. code-block:: text
+
+    <s>[INST] <<SYS>>
+    {{ system_prompt }}
+    <</SYS>>
+
+    {{ user_message_1 }} [/INST] {{ model_answer_1 }} </s>
+    <s>[INST] {{ user_message_2 }} [/INST]
```

Comment:
This is nice but you need to give actual examples. Take a specific input text and walk through all the steps: prompt format, tokenization, special tokens, etc. Give a consistent simple example that readers can always anchor back to for their understanding.